examples/dialogs/dialogs/app.py

```diff
@@ -90,12 +90,53 @@ def action_save_file_dialog(self, widget):
         except ValueError:
             self.label.text = "Save file dialog was canceled"
 
+    def action_open_secondary_window(self, widget):
+        # Problems:
+        # - Setting secondary window's `app` manually
+        # - Secondary window's `icon` is not set
+        def close_handler(win):
+            self.window_count -= 1
+            self.set_window_label_text()
+        window = toga.Window(title="New Window {}".format(self.window_count))
+        self.window_count += 1
+        self.set_window_label_text()
+        secondary_label = toga.Label(text="You are in secondary window!")
+        window.content = toga.Box(
+            children=[
+                secondary_label
+            ],
+            style=Pack(
+                flex=1,
+                direction=COLUMN,
+                padding=10
+            )
+        )
+        window.app = self
+        window.on_close = close_handler
+        window.show()
+
+    def exit_handler(self, app, cancel_exit):
+        if self.main_window.confirm_dialog('Toga', 'Are you sure you want to quit?'):
+            print("Label text was \'{}\' when you quit the app".format(self.label.text))
+        else:
+            cancel_exit()
+            self.label.text = 'Exit canceled'
+
+    def set_window_label_text(self):
+        self.window_label.text = '{} secondary windows open'.format(self.window_count - 1)
+
     def startup(self):
         # Set up main window
         self.main_window = toga.MainWindow(title=self.name)
+        self.window_count = 1
```

Comment:
It might be better to keep this as a "secondary window count" - that way we can avoid doing the "and subtract one" every time we actually use it.

Reply:
Great point!